### ISSUE TITLE
cloud-provider broke the ability to drain the node

### ISSUE THREAD

**Issue by nickngch** [rancher/rancher#28589]
**What kind of request is this (question/bug/enhancement/feature request):**
BUG

**Steps to reproduce (least amount of steps as possible):**
1. Create a create with cloud provider enabled(aws in this case)
2. Try to drain a node or enable node drain during cluster upgrade

**Result:**
Node failed to drain, the node status stuck in draining

**Other details that may be helpful:**
It looks like the cloud provider is replacing the node name to `ip-172-31-39-209.us-east-2.compute.internal` while the Rancher is trying to drain the node with `ip-172-31-39-209`

kubelet logs:
```
W0827 01:35:26.912391    9849 setters.go:183] replacing cloudprovider-reported hostname of ip-172-31-39-209.us-east-2.compute.internal with overridden hostname of ip-172-31-39-209
W0827 01:35:36.943886    9849 setters.go:183] replacing cloudprovider-reported hostname of ip-172-31-39-209.us-east-2.compute.internal with overridden hostname of ip-172-31-39-209
```

Rancher logs:
```
2020/08/27 02:08:04 [ERROR] nodeDrain: kubectl error draining node [ip-172-31-39-209] in cluster [c-l856v]:
2020/08/27 02:08:04 [INFO] Draining node ip-172-31-39-209 in c-l856v with flags --delete-local-data=false --force=false --grace-period=-1 --ignore-daemonsets=true --timeout=120s
2020/08/27 02:08:04 [ERROR] nodeDrain: kubectl error draining node [ip-172-31-39-209] in cluster [c-l856v]:
2020/08/27 02:08:04 [INFO] Draining node ip-172-31-39-209 in c-l856v with flags --delete-local-data=false --force=false --grace-period=-1 --ignore-daemonsets=true --timeout=120s
2020/08/27 02:08:04 [ERROR] nodeDrain: kubectl error draining node [ip-172-31-39-209] in cluster [c-l856v]:
2020/08/27 02:08:04 [INFO] Draining node ip-172-31-39-209 in c-l856v with flags --delete-local-data=false --force=false --grace-period=-1 --ignore-daemonsets=true --timeout=120s
2020/08/27 02:08:04 [ERROR] nodeDrain: kubectl error draining node [ip-172-31-39-209] in cluster [c-l856v]:
```

 k get node
```
NAME                                          STATUS   ROLES                      AGE   VERSION
ip-172-31-39-209.us-east-2.compute.internal   Ready    worker                     18m   v1.15.12
ip-172-31-45-250.us-east-2.compute.internal   Ready    controlplane,etcd,worker   21m   v1.16.13
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.4.5
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 2/2
- Kubernetes version (use `kubectl version`):

```
all verions
```

- Docker version (use `docker version`):

```
Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:45:36 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:44:07 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

gz#11932

**Comment by superseb** [comment]
This is most likely resolved with https://github.com/rancher/rancher/issues/23333

**Comment by nickngch** [comment]
@superseb Thanks for the information, the issue has fixed in Rancher v2.4.6. Closing this ticket.
